apps/api/src/modules/providers/push-sns/push-sns.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@nestjs/common';
+import { SNS } from 'aws-sdk';
+import { ProvidersService } from '../providers.service';
+
+export interface PushSnsData {
+  target: string;
+  title: string;
+  body: string;
+}
+
+@Injectable()
+export class PushSnsService {
+  private sns: SNS;
+
+  constructor(private readonly providersService: ProvidersService) {}
+
+  async assignSnsConfig(providerId: number): Promise<void> {
+    const snsConfig = await this.providersService.getConfigById(providerId);
+
+    this.sns = new SNS({
+      accessKeyId: snsConfig.AWS_ACCESS_KEY_ID as string,
+      secretAccessKey: snsConfig.AWS_SECRET_ACCESS_KEY as string,
+      region: snsConfig.AWS_REGION as string,
+    });
+  }
+
+  async sendPushNotification(data: PushSnsData, providerId: number): Promise<SNS.PublishResponse> {
+    await this.assignSnsConfig(providerId);
+    const { title, body } = data;
+    const message = `${title}: ${body}`;
```

Comment:
DId you see how you are getting notification with FCM when we use title and body?
Here from what I see we have kept title and body as necessary but then you are fixing the format of notification limiting the usecase of notification message.

Once check what effect title and body does from firebase. Based on your findings, we can just keep message as the field and remove title and body if they don't effect.

Here I see message as an object, you can test with this in SNS too and see its formatted correctly.
![image](https://github.com/OsmosysSoftware/osmo-x/assets/14059393/8b1ba8d2-04e0-4143-8153-70b33406f0c5)

Reply:
updated in https://github.com/OsmosysSoftware/osmo-x/pull/269/commits/c89ab05d02d57df395c3776d0070d1823d4712cd